handlers/SendReminder.ts

```diff
@@ -0,0 +1,209 @@
+    const timeDifference = currentDate.getTime() - createdAtDate.getTime();
+    const daysOld = Math.floor(timeDifference / (1000 * 60 * 60 * 24));
+    pr.ageInDays = daysOld;
+  }
+
+  pullRequestsWaitingForReview.sort((a, b) => (a.ageInDays || 0) - (b.ageInDays || 0));
+
+  const Pulls = pullRequestsWaitingForReview.length;
+
+  const appUser = await read.getUserReader().getAppUser() as IUser;
+  const room = await getDirect(read, modify, appUser, username) as IRoom;
+
+  const textSender = modify.getCreator().startMessage();
+
+  if (Pulls > 0) {
+    textSender.setText(`🚀 You've got ${Pulls} pull requests waiting for your review. Give them the green light 💚`);
+  } else {
+    textSender.setText(`📅 It appears that there are no pull requests for you to review today. Enjoy your PR-free day! 🚀`);
+  }
+
+  if (room) {
+    textSender.setRoom(room);
+  }
+
+  await modify.getCreator().finish(textSender);
+
+  const modifyCreator = modify.getCreator();
+
+  for (const [ind, pull] of pullRequestsWaitingForReview.entries()) {
+    let markdownText = `${pull.repo} #${pull.number}\n`;
+
+    if (pull.ageInDays && pull.ageInDays > 0) {
+      markdownText += `*${pull.ageInDays} days old*\n`;
+    }
+
+    markdownText += `**[${pull.title}](${pull.url})**`;
+
+    const block = modifyCreator.getBlockBuilder();
+
+    block.addSectionBlock({
+      text: block.newMarkdownTextObject(markdownText)
+    });
+
+    block.addActionsBlock({
+      blockId: "githubdata",
+      elements: [
+        block.newButtonElement({
+          actionId: "unsubscribe-reminder-action",
+          text: block.newPlainTextObject("Unsubscribe"),
+          value: `${pull.repo}|${pull.number}`,
+          style: ButtonStyle.DANGER,
+        })
+      ],
+    });
+
+    if (ind < 10) {
+      const textSender = await modifyCreator.startMessage().setBlocks(block);
+      if (room) {
+        textSender.setRoom(room);
+      }
+      await modifyCreator.finish(textSender);
+    }
+  }
+}
+
+async function CheckIsReviewRequested(githubusername: string, pr: any): Promise<boolean> {
+  const reviewers = pr.requested_reviewers;
+
+  for (const reviewer of reviewers) {
+    if (reviewer.login === githubusername) {
+      return true;
+    }
+  }
+  return false;
+}
+
+async function CheckIsUserInTeam(githubusername: string, access_token: string | undefined, http: IHttp, pr: any): Promise<boolean> {
+  if (pr.requested_teams.length === 0) {
+    return false;
+  }
+
+  const teams = pr.requested_teams;
+
+  for (const team of teams) {
+    const response = await http.get(`https://api.github.com/teams/${team.id}/members/${githubusername}`, {
+      headers: {
+        Authorization: `token ${access_token}`,
+        "Content-Type": "application/json",
+      },
+    });
+
+    if (!response.statusCode.toString().startsWith("2")) {
+      return false;
+    }
+
+    if (response.statusCode === 204) {
+      return true;
+    }
+  }
+
+  return false;
```

Comment:
This whole code is for sending reminders form github app,
Please don't copy paste code without understanding it.

Reply:
@RyanbowZ was this addressed ?